File: 프로그래머스/Lv.1/12977. 소수 만들기/소수 만들기.py
from itertools import combinations

def solution(nums):
    c = combinations(nums, 3)
    count = 0
    
    for i in c:
        if is_prime(sum(i)):
            count += 1
    
    return count

def is_prime(n):
    for i in range(2, n):
        if n % i == 0:
            return False
    
    return True

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.java
import java.util.*;

class Solution {
    private final int r = 3;
    private int count = 0;
    private int[] picked;
    
    public int solution(int[] nums) {
        picked = new int[r];
        backTracking(0, 0, nums);

        return count;
    }
    
    private void backTracking(int at, int depth, int[] nums) {
        if (depth == r) {
            int sum = (int)Arrays.stream(picked).sum();
            
            if (isPrime(sum)) count++;
            
            return;
        }
        
        for (int i = at; i < nums.length; i++) {
            picked[depth] = nums[i];
            backTracking(i + 1, depth + 1, nums);
        }
    }
    
    private boolean isPrime(int n) {
        if (n == 2) return true;
        
        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (n % i == 0) return false;
        }
        
        return true;
    }
}